config.go

```diff
@@ -239,5 +246,12 @@ func (cfg Config) openSinks() (zapcore.WriteSyncer, zapcore.WriteSyncer, error)
 }
 
 func (cfg Config) buildEncoder() (zapcore.Encoder, error) {
+	if cfg.EncoderConfig.TimeKey != "" && cfg.EncoderConfig.EncodeTime == nil {
+		cfg.EncoderConfig.EncodeTime = zapcore.EpochTimeEncoder
```

Comment:
Same here. It's not obvious why we pick epoch for production and ISO8601 for development. Perhaps an error would be preferable?

Reply:
NewDevelopmentEncoderConfig specifies ISO8601TimeEncoder by default. I have not made any explicit choices here. But, let's return an error.